crates/extensions/tedge_mqtt_ext/src/lib.rs

```diff
@@ -115,6 +122,7 @@ impl FromPeers {
         outgoing_mqtt: &mut mpsc::UnboundedSender<MqttMessage>,
     ) -> Result<(), RuntimeError> {
         while let Ok(Some(message)) = self.try_recv().await {
+            tracing::info!(target: "MQTT pub", "[{}] {}", &message.topic.name, message.payload_str().unwrap_or_default());
```

Comment:
Now a message that is published is logged as:

```
2024-06-26T15:35:33.01872567Z  INFO MQTT pub: [te/device/main///cmd/software_list/c8y-mapper-2024-06-26T15:35:33.010468866Z] {"status":"init"}
```

This has to be compared to the current log entries for the same:

```
2024-06-20T17:50:27.811717059Z DEBUG MQTT: recv Ok(Some(MqttMessage { topic: Topic { name: "te/device/main///cmd/software_list/c8y-mapper-2024-06-20T17:50:27.811587059Z" }, payload: "{"status":"init"}", qos: AtLeastOnce, retain: true })) 
```

Which is confusing. This has to be read as "the MQTT actor receive a request to publish a message".
With the new proposal, it's clear: the message is published over MQTT.

That said, the following has to be discussed:
- Is INFO the right level?  This is really helpful for commands, but might be overwhelming with telemetry data.
- One can configure the level depending on the topic. Is it worth it?
- Do we have to log QoS and Retain flags?

Reply:
> Is INFO the right level? This is really helpful for commands, but might be overwhelming with telemetry data.
I would default to `DEBUG` level for MQTT message logging. We can have additional command handling logging at `INFO` level as the message is handled in context of the workflows (except for child devices with external agents doing their own custom command processing).

> One can configure the level depending on the topic. Is it worth it?
No I don't think so either, though as long as the first comment above can be addressed.

> Do we have to log QoS and Retain flags?
Yes, QoS and Retain flags would be useful, though if possible I would like to avoid logging of Rust related stuff e.g. `Ok(Some(MqttMessage` seems unnecessary to me.

Reply:
Improved by https://github.com/thin-edge/thin-edge.io/pull/2964/commits/0b497e6d238ae8f1229420c2fe20f09cc42d9a76